pkg/openai/main.go

```diff
@@ -231,6 +227,31 @@ func (c *Connection) Execute(inputs []*connectorPB.DataPayload) ([]*connectorPB.
 				},
 			})
 		}
+	case speechRecognitionTask:
+		for i, dataPayload := range inputs {
+			noOfAudios := len(dataPayload.Audios)
+			if noOfAudios <= 0 {
+				return inputs, fmt.Errorf("no audios given")
+			}
+			req := AudioTranscriptionReq{
+				File:        dataPayload.Audios[0],
+				Model:       c.getModel(),
+				Language:    dataPayload.GetMetadata().GetFields()["language"].GetStringValue(),
+				Temperature: dataPayload.GetMetadata().GetFields()["temperature"].GetNumberValue(),
+			}
+			if len(dataPayload.Texts) > 0 {
+				req.Prompt = dataPayload.Texts[0]
+			}
+			resp, err := client.GenerateAudioTranscriptions(req)
+			if err != nil {
+				return inputs, err
+			}
+			outputs = append(outputs, &connectorPB.DataPayload{
+				DataMappingIndex: inputs[i].DataMappingIndex,
+				Texts:            []string{resp.Text},
+				Audios:           dataPayload.Audios,
```

Comment:
We don't need to return the input audio in the output

Reply:
changed